### PR TITLE
Single order page / update payment order

### PR DESCRIPTION
**Pull request by falklinus**
# Jira Issue: [CLX-258]

## What?
- Lots of refactoring for payment form, making it usable for edit
- Add page for "single order" and ability to route directly to it
- Add section to edit order in the single order page

## Why?
- Edit order
- Link to specific orders

## Optional screenshots
image

## Optional checklist
- [ ] Updated `src/changelog.ts`
- [x] Codescouted
- [ ] Unit tests written
- [x] Tested locally

[CLX-258]: https://hedvig.atlassian.net/browse/CLX-258?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/ManualPaymentFormFields.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  UseFormWatch,
+} from 'react-hook-form'
+
+interface Props {
+  recipientType: string
+  memberFullName: string
+  watch: UseFormWatch<FieldValues>
+  setValue: UseFormSetValue<any>
+  register: UseFormRegister<any>
+  method: string
```
[the remark this answers is not in the document]
Comment (falklinus) [reply src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/ManualPaymentFormFields.tsx:29]: I think so, should complain where you consume it if it doesn't

File: src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/SubclaimPaymentOrderForm.tsx [hunk 2]
```diff
@@ -79,354 +69,41 @@ const SubclaimPaymentOrderForm: React.FC<{
...
+  order?: PaymentOrderInformationFragment
+  onSuccess?: () => void
+  subclaimId: string
+  memberId?: string
+  memberFullName: string
+  preferredCurrency?: string
+  isPaymentActivated: boolean
+  isPotentiallySanctioned: boolean
+  defaultDeductible: string
+  carrier?: string
```
[the remark this answers is not in the document]
Comment (falklinus) [reply src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/SubclaimPaymentOrderForm.tsx:103]: Yes, should probably use it

File: src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/SubclaimPaymentOrdersTable.tsx [hunk 3]
```diff
@@ -278,10 +279,7 @@ const SubclaimPaymentOrdersTable: FC<{
                   </TableColumn>
                   <TableColumn style={{ width: '7rem' }}>
                     {!nextState ? (
-                      <Tag
-                        status={tagStatus(state)}
-                        style={{ cursor: 'not-allowed' }}
```
[the remark this answers is not in the document]
Comment (falklinus) [reply src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/SubclaimPaymentOrdersTable.tsx:283]: I think it is clear enough that nothing happens when you hover over it, but i'm not a UX expert 🤪

File: src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/UpsertPaymentOrder.tsx [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+          bic: '',
+          note: '',
+          reference: '',
+        }
+      : {
+          recipientName: order.recipientName,
+          deductible: order.deductible.amount.toString(),
+          beforeAmount: (
+            order.amount.amount + order.deductible.amount
+          ).toString(),
```
[the remark this answers is not in the document]
Comment (falklinus) [reply src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/UpsertPaymentOrder.tsx:68]: I think it is ok to have it in the function

File: src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/UpsertPaymentOrder.tsx [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+    recipientType,
+    memberFullName,
+    isManualPayment,
+    isPaymentActivated,
+  ])
+
+  const clearForm = (
+    updated: PaymentOrderInformationFragment | null = null,
+  ) => {
+    const form = updated ?? order
```
[the remark this answers is not in the document]
Comment (falklinus) [reply src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/UpsertPaymentOrder.tsx:171]: maybe just updatedOrder is more clear 🤷

File: src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/UpsertPaymentOrder.tsx [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+    if (form) {
+      setIsExGratia(form.isExGratia)
+      setSanctionListSkipped(!!form.sanctionListSkipped)
+      setIsManualPayment(form.method !== PaymentMethod.AUTOGIRO)
+      setRecipientType(form.recipientType)
+      setDueDate(form.dueDate)
+      setReadyInBank(false)
```
[the remark this answers is not in the document]
Comment (falklinus) [reply src/portals/hope/features/claims/claim-details/SubclaimPaymentOrders/components/UpsertPaymentOrder.tsx:178]: Not sure if we want it checked once we submit the order?

**Review by vonElfvin — APPROVED** [review]

**Pull request merged**